code/datums/cleaning.dm

```diff
@@ -0,0 +1,52 @@
+//For handling standard click-to-clean items like soap and mops.
+
+//text1, text2, and text3 carry strings that, when pieced together by this proc, make the cleaning messages.
+//text1 and text2 pertain to the start message, i.e. "User begins to [clean] the floor[ with the mop.]"
+//text3 pertains to the end message, i.e. "You [finished mopping] the floor."
+
+/atom/proc/cleaning_act(mob/user, atom/cleaner, cleanspeed = 50, text1 = "clean", text2 = " with [cleaner].", text3 = "clean")
```

Comment:
so one other things for these parameters, instead of using `text1`, `text2`, .... and so on. Could you give each one unique names pertaining to their purpose in messages, so something like `end_text` instead of `text3`

That and your documentation will make it way easier for future devs to understand this proc.